controllers/kubeadmbootstrapconfig_controller.go

```diff
@@ -42,11 +44,28 @@ func (r *KubeadmBootstrapConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Res
 	ctx := context.Background()
 	log := r.Log.WithValues("kubeadmbootstrapconfig", req.NamespacedName)
 
-	machine := &clusterapiv1alpha2.Machine{}
-	if err := r.Get(ctx, req.NamespacedName, machine); err != nil {
+	config := kubeadmv1alpha1.KubeadmBootstrapConfig{}
+	if err := r.Get(ctx, req.NamespacedName, &config); err != nil {
 		log.Error(err, "stacktrace", fmt.Sprintf("%+v", err))
+		return ctrl.Result{}, errors.WithStack(err)
 	}
 
+	// Find the owner reference
+	machineKind := v1alpha2.SchemeGroupVersion.WithKind("Machine").String()
+	var machineRef v1.OwnerReference
+	for _, ref := range config.OwnerReferences {
+		if ref.Kind == machineKind {
+			machineRef = ref
+			break
+		}
+	}
+	if machineRef.Kind == "" {
```

Comment:
non-blocking: alternatively machineRef could have been a pointer.

Reply:
oh and then checked for nil? That's a good idea!